### ISSUE TITLE
Make Pillow Opt-in

### ISSUE THREAD

**Issue by cgarciae** [tflearn/tflearn#162]
Hi, I am maintaining TensorBuilder and I port a couple of you functions into the library. However, TensorBuilder has become "harder than necessary" to install because of TFLearn's dependency on Pillow. It would be nice if Pillow was optional, giving an error message if you try to use a TFLearn module that depends on Pillow.

**Comment by cgarciae** [comment]
The thing with Pillow is that it depends on a couple of native Linux libraries like `jpeg`. Users become a little confused because

1. They might not know TensorBuilder uses `tflearn`, although it is mentioned somewhere.
2. Its even less probable that they know what `pillow` is.

So they don't understand the error and its generally a bad experience when just doing an import statement.

BTW:  TensorBuilder's `core` literally is just a layer that enables the use of nice syntax, to the point I actually don't code the operations that deal with Tensors myself, I register functions from various libraries at runtime (I do preserve the original documentation and clarify what module it comes from). I've done it in such a way that, if you so desired, you could **patch** an untainted `Builder` class with only TFLearn functions and you get the DSL for free. My approach for TensorBuilder as a useful library is to give a curated patch with useful functions from any library, I might make the core a separate library to make the separation more clear. I think I'll be sponsored to improve the library.

**Comment by aymericdamien** [comment]
True, so it is a good idea to make Pillow optional in our next update.
